docs/index.rst

```diff
@@ -268,6 +268,35 @@ then the *hookimpl* should be marked with the ``"optionalhook"`` option:
 
         return config
 
+.. _specname:
+
+Hook implementation naming
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+During plugin :ref:`registration <registration>`, hook implementations are
+matched to a hook :ref:`specifications <specs>` in the *host* program with the
+same name as the function being decorated by ``@hookimpl`` (e.g.
+``setup_project`` in the example above).
+
+*new in version 0.13.2:*
+
+To override the default behavior, a *hookimpl* may also be matched to a
+*hookspec* in the ``host program`` with a non-matching name by using the
+``specname`` option.  Continuing the example above, the *hookimpl* function
+does not need to be named ``setup_project``:
```

Comment:
does not need to be named ``setup_project`` but if `specname="setup_project"` it will be matched and checked against the `setup_project` *hookspec*.